features/manage-position/SettleExpired.tsx

```diff
@@ -0,0 +1,297 @@
+import { useState } from "react";
+import styled from "styled-components";
+import { Box, Button, Typography, Grid } from "@material-ui/core";
+import { utils } from "ethers";
+
+import EmpContract from "../../containers/EmpContract";
+import EmpState from "../../containers/EmpState";
+import Collateral from "../../containers/Collateral";
+import Position from "../../containers/Position";
+import DvmState from "../../containers/DvmState";
+import Token from "../../containers/Token";
+import Etherscan from "../../containers/Etherscan";
+import { DOCS_MAP } from "../../utils/getDocLinks";
+
+const Link = styled.a`
+  color: white;
+  font-size: 14px;
+`;
+
+const Important = styled(Typography)`
+  color: red;
+  background: black;
+  display: inline-block;
+`;
+
+enum CONTRACT_STATE {
+  OPEN,
+  PRICE_REQUESTED,
+  PRICE_RECEIVED,
+}
+
+const { parseBytes32String: hexToUtf8 } = utils;
+
+const SettleExpired = () => {
+  const { contract: emp } = EmpContract.useContainer();
+  const { empState } = EmpState.useContainer();
+  const {
+    priceIdentifier,
+    expirationTimestamp,
+    contractState,
+    isExpired,
+  } = empState;
+  const { symbol: collSymbol } = Collateral.useContainer();
+  const { tokens: posTokens, collateral: posColl } = Position.useContainer();
+  const {
+    symbol: tokenSymbol,
+    allowance: tokenAllowance,
+    decimals: tokenDec,
+    setMaxAllowance,
+    balance: tokenBalance,
+  } = Token.useContainer();
+  const { getEtherscanUrl } = Etherscan.useContainer();
+  const { dvmState } = DvmState.useContainer();
+  const { hasEmpPrice, resolvedPrice } = dvmState;
+
+  const [hash, setHash] = useState<string | null>(null);
+  const [success, setSuccess] = useState<boolean | null>(null);
+  const [error, setError] = useState<Error | null>(null);
+
+  if (
+    posTokens !== null &&
+    posColl !== null &&
+    tokenBalance !== null &&
+    tokenDec !== null &&
+    tokenSymbol !== null &&
+    tokenAllowance !== null &&
+    emp !== null &&
+    priceIdentifier !== null &&
+    expirationTimestamp !== null &&
+    contractState !== null &&
+    hasEmpPrice !== null &&
+    isExpired !== null &&
+    isExpired // If contract has not expired, then do not render this component
+  ) {
+    const priceIdUtf8 = hexToUtf8(priceIdentifier);
+    const expiryDate = new Date(
+      Number(expirationTimestamp) * 1000
+    ).toLocaleString("en-GB", { timeZone: "UTC" });
+
+    const needsToRequestSettlementPrice = contractState === CONTRACT_STATE.OPEN;
+
+    // Error conditions for calling settle expired:
+    const needAllowance =
+      tokenAllowance !== "Infinity" && tokenAllowance < tokenBalance;
+    const canSettleTokens = tokenBalance > 0 || posTokens > 0;
+
+    // Calculate collateral to receive if price was resolved.
+    let positionTRV: number = -1;
```

Comment:
this is a new pattern. why are you setting the defaults to -1?

Reply:
you'er right, I ended up replacing these as `let positionTRV: number;`

Reply:
Which, I'm surprised you can do with TS

Reply:
Nevermind, not possible. Set them all to type `number | null` and have to check for null lower down where I render them